diff --git a/webform_mailchimp.module b/webform_mailchimp.module
index 6a1f24c..f4cee6f 100644
--- a/webform_mailchimp.module
+++ b/webform_mailchimp.module
@@ -132,9 +132,9 @@ function webform_mailchimp_webform_submission_insert($node, $submission) {
         }
       }
       foreach ($groupfields_replacements as $groupname => $values_array) {
-        $mergefields_replacements['GROUPINGS'][] = array(
+        $mergefields_replacements['groupings'][] = array(
           'name' => $groupname,
-          'groups' => implode(',', $values_array),
+          'groups' => $values_array,
         );
       }
     }
